cores/arduino/SERCOM.cpp

```diff
@@ -135,6 +135,11 @@ bool SERCOM::isUARTError()
   return sercom->USART.INTFLAG.bit.ERROR;
 }
 
+bool SERCOM:: isParityOrFrameError()
+{
+  return sercom->USART.STATUS.bit.FERR | sercom->USART.STATUS.bit.PERR ;
```

Comment:
here i think we can optimize this by doing a single read of `.reg` instead of `.bit` and comparing the bit mask